### PR TITLE
BEMINFO-563 - bem-info: Language switch has been fallen

### PR DESCRIPTION
**Pull request by gela-d**

### CODE REVIEW

File: src/blocks/common.blocks/lang-switch/lang-switch.styl [hunk 1]
```diff
@@ -1,19 +1,12 @@
-.lang-switch
+.lang-switch .link
```
Comment (tadatuta) [inline src/blocks/common.blocks/lang-switch/lang-switch.styl]: isn't it possible to use mix instead of cascade here?

Comment (tavriaforever) [inline src/blocks/common.blocks/lang-switch/lang-switch.styl]: Bring an example code

Comment (tavriaforever) [inline src/blocks/common.blocks/lang-switch/lang-switch.styl]: oh)) I understood your proposition. You right, mix element would be much better

File: src/blocks/common.blocks/lang-switch/lang-switch.bemhtml [hunk 2]
```diff
@@ -1,9 +1,10 @@
...
-        block: 'link',
-        url: this.ctx.url,
-        content: BEM.I18N('lang-switch', 'name')
-    }
-}
+block('lang-switch')(
+    def()(function() {
+        return applyCtx({
+            block: 'link',
+            mix: [{ block: this.block }],
```
Comment (gela-d) [inline src/blocks/common.blocks/lang-switch/lang-switch.bemhtml:5]: @tadatuta Here is some problem. When we write this mix without square brackets, our bem-info site is fallen down.
This is bug, but we don't understand where exactly

Comment (tadatuta) [inline src/blocks/common.blocks/lang-switch/lang-switch.bemhtml:5]: just created testcase:

``` js
block('b1').def()(function() {
    return applyCtx({
        block: 'bla',
        mix: { block: this.block }
    });
});
```

and it works just fine.

the bug is somewhere in templates of `bem-site-engine`.

can you at least paste error log here?

Comment (gela-d) [inline src/blocks/common.blocks/lang-switch/lang-switch.bemhtml:5]: @tadatuta error log from building bem-info with this case - 
[2014-10-09 13:28:24] ERROR __error.middleware__error.node.js: 500 Unknown error
And my page looks like

Comment (gela-d) [inline src/blocks/common.blocks/lang-switch/lang-switch.bemhtml:5]: @tadatuta this case is

Comment (tadatuta) [inline src/blocks/common.blocks/lang-switch/lang-switch.bemhtml:5]: i understand that the error is somewhere is templates ;)
so i'd be great to have logs from stderr )

**Comment by gela-d** [comment]
@tadatuta i merged this pr, because all of problems solved.
But I want to know why this mix isn't work without array. We found that, if we add {block: this.block, **elem: 'bla'**}, our mix works with object brackets.

**Pull request merged**